### ISSUE TITLE
Issues due to faulty tscn and escn files

### ISSUE THREAD

**Issue by sdfgeoff** [godotengine/godot#17372]
I'm working on the blender exporter, and have discovered that I can make godot crash quite easily. Rather than open lots of bug reports, I thought I'd collate them here. I will likely add more over the next few days/weeks as I isolate the conditions that cause issues.

I consider these to be relatively low priority. Yes, many can crash Godot, but most of the time Godot is working with the files it exports (and it exports them correctly). 

Issues:
 - [x] Segfaults if first node has a parent (ie: no scene root)
 - [ ] Segfaults if an an `.escn` is edited externally and gets reimported while the escn is visible (ie inherited scene or "open anyway"). If it doesn't segault, the import is often poor with incorrect transformations.
 - [x] Segfaults if vectors/color missing a component (eg `Color ( 1, 1, 1 )` or `Vector3( 0, 0, )`)

**Comment by beniwtv** [comment]
I also have an issue with an .escn file generated with the Blender exporter. If the model is very complex, lots of geometry details is put into the file and Godot fails to read it correctly (seems like some line length limit?).

Godot then complains about the syntax of the file, which is however correct as far as I can see.

**Comment by sdfgeoff** [comment]
Hmm. That could probably be solved quite easily (but ideally temporarily) in the exporter if you need a solution fast.

**Comment by sdfgeoff** [comment]
@aaronfranke 
It should throw an error on import. Something like:
```
Faulty ESCN File: line 13
```
Under no situation should a program _ever_ segfault.

**Comment by Calinou** [comment]
I can reproduce this in 3.2.2beta 4. Save the file below as `test.tscn` and open it in the Godot editor:

```ini
[gd_scene format=2]

[node name="Node2D" type="Node2D"]
position = Vector2( 20 )
```

**Comment by akien-mga** [comment]
>     * Segfaults if first node has a parent (ie: no scene root)

Fixed by #39389.

>     * Segfaults if vectors/color missing a component (eg `Color ( 1, 1, 1 )` or `Vector3( 0, 0, )`)

Fixed by #39386.

>     * Segfaults if an an .escn is edited externally and gets reimported while the escn is visible (ie inherited scene or "open anyway"). If it doesn't segault, the import is often poor with incorrect transformations.

I suggest opening a dedicated issue for this with a MRP if it's still reproducible (generally-speaking, all three issues should have had their own ticket with an MRP as they're not related to each other).
